### PR TITLE
Apply shared checkstyles to all samples.

### PR DESCRIPTION
**Pull request by tswast**
We use [subtree merging](https://git-scm.com/book/en/v1/Git-Tools-Subtree-Merging) to
pull these from the [java-repo-tools](https://github.com/GoogleCloudPlatform/java-repo-tool) repository. That way we keep a copy of the needed files within this repo and `git clone` still works.

I ran the following commands:

```
git remote add java-repo-tools git@github.com:GoogleCloudPlatform/java-repo-tools.git
git fetch java-repo-tools master
git checkout -b java-repo-tools java-repo-tools/master
git checkout master
git checkout -b update-java-repo-tools
git read-tree --prefix=java-repo-tools/ -u java-repo-tools
```

and then updated the main POM to use the one from `java-repo-tools`.

I've filed https://github.com/GoogleCloudPlatform/java-repo-tools/issues/5 to document this in the README.

### CODE REVIEW

**Comment by tswast** [comment]
@lesv Could you take a look.

Using a subtree means the files are just copied here, basically, so a regular clone still works. I will have to document how to pull updates from upstream, yet, but I think I can automate that in the same way we plan to automate the maven updates.

File: pom.xml [hunk 1]
```diff
@@ -8,10 +8,17 @@
     <artifactId>doc-samples</artifactId>
     <packaging>pom</packaging>
 
+    <!-- Parent POM defines common plugins and properties. -->
```
Comment (lesv) [inline pom.xml]: This should get a license at the top and if you wish, you can get rid of the xmlns / xsi stuff.

File: pom.xml [hunk 2]
```diff
@@ -8,10 +8,17 @@
...
+    <parent>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>shared-configuration</artifactId>
+        <version>1.0.0</version>
+        <relativePath>java-repo-tools</relativePath>
+    </parent>
+
```
Comment (lesv) [inline pom.xml]: 1.9.34 was pushed last night.

**Comment by lesv** [comment]
Nits - LGTM

File: pom.xml (hunk 2 above)
Comment (tswast) [inline pom.xml]: I mailed https://github.com/GoogleCloudPlatform/java-docs-samples/pull/129 to do this in a separate PR.
